src/domain.py

```diff
@@ -228,12 +229,94 @@ def build_juju_unit_status(pod_status):
     return unit_status
 
 
-def build_prometheus_config(external_labels, advertised_port, monitor_k8s):
+def validate_and_parse_external_labels(raw_labels):
+    """
+    Determines, if the input variable can be safely injected into the
+    Prometheus config (input variable should be a dict, containing ONLY the
+    strings as its values).
+
+    :param raw_labels: Charm 'external-labels' config option.
+    """
+    ERROR_MESSAGE = "external-labels malformed JSON"
+
+    if not raw_labels:  # empty string
+        return {}
+
+    try:
+        parsed_labels = json.loads(raw_labels)
+    except (ValueError, TypeError):
+        raise Error("{0}: {1}".format(ERROR_MESSAGE, raw_labels))
+
+    if not isinstance(parsed_labels, dict):
```

Comment:
Have you considered using JSON Schema for this part?

Reply:
Well no, I tried to avoid introducing an external dependency here. Do you think it's reasonable to introduce it here?

Reply:
Yeah good point. Let's drop that for now.